tests/nativeapp/test_package_scripts.py

```diff
@@ -50,148 +54,169 @@ def _get_na_manager(working_dir):
     )
 
 
-@mock.patch(SQL_EXECUTOR_EXECUTE_QUERIES)
-@mock.patch(SQL_EXECUTOR_EXECUTE)
+def use_project_with_package_scripts():
+    package_script_1 = dedent(
+        """\
+        -- package script (1/2)
+        create schema if not exists {{ package_name }}.my_shared_content;
+        grant usage on schema {{ package_name }}.my_shared_content
+        to share in application package {{ package_name }};
+        """
+    )
+    package_script_2 = dedent(
+        """\
+        grant select on table {{ package_name }}.my_shared_content.shared_table
+          to share in application package {{ package_name }};
+        """
+    )
+
+    ProjectV10Factory(
+        pdf__native_app__name="myapp",
+        pdf__native_app__package__name="myapp_pkg_polly",
+        pdf__native_app__artifacts=["setup.sql"],
+        pdf__native_app__package__scripts=["001-shared.sql", "002-shared.sql"],
+        files={"001-shared.sql": package_script_1, "002-shared.sql": package_script_2},
+    )
+
+    rendered_script_1 = dedent(
+        """\
+        -- package script (1/2)
+        create schema if not exists myapp_pkg_polly.my_shared_content;
+        grant usage on schema myapp_pkg_polly.my_shared_content
+        to share in application package myapp_pkg_polly;
+        """
+    )
+    rendered_script_2 = dedent(
+        """\
+        grant select on table myapp_pkg_polly.my_shared_content.shared_table
+          to share in application package myapp_pkg_polly;
+        """
+    )
+    return {"001-shared.sql": rendered_script_1, "002-shared.sql": rendered_script_2}
+
+
 @mock_connection()
+def test_package_scripts_with_conn_warehouse(
+    mock_conn,
+    temp_dir,
+):
+    scripts = use_project_with_package_scripts()
+    with mock.patch.object(
+        SnowflakeSQLFacade, "execute_user_script"
+    ) as mock_execute_user_script:
+        mock_execute_user_script.return_value = None
+        native_app_manager = _get_na_manager(str(temp_dir))
+        native_app_manager._apply_package_scripts()  # noqa: SLF001
+        expected_calls = [
+            mock.call(query[1], query[0], "role", "wh")
+            for query in list(scripts.items())
+        ]
+        assert mock_execute_user_script.mock_calls == expected_calls
+
+
+@mock_connection()
+def test_package_scripts_with_pdf_warehouse(
+    mock_conn,
+    temp_dir,
+):
+    scripts = use_project_with_package_scripts()
+    PdfV10Factory.with_filename("snowflake.local.yml")(
+        native_app__package__warehouse="myapp_pkg_warehouse"
+    )
+    with mock.patch.object(
+        SnowflakeSQLFacade, "execute_user_script"
+    ) as mock_execute_user_script:
+        mock_execute_user_script.return_value = None
+        native_app_manager = _get_na_manager(str(temp_dir))
+        native_app_manager._apply_package_scripts()  # noqa: SLF001
+        assert mock_execute_user_script.call_count == 2
```

Comment:
sorry, I thought "and use the same pattern everywhere else" was implied

Reply:
yes that was my bad I missed that there was more instances. fixed.